apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/JvmMemoryMetrics.java

```diff
@@ -26,56 +26,91 @@
 import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.matcher.WildcardMatcher;
 import co.elastic.apm.agent.metrics.DoubleSupplier;
 import co.elastic.apm.agent.metrics.Labels;
 import co.elastic.apm.agent.metrics.MetricRegistry;
 
 import java.lang.management.ManagementFactory;
 import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryType;
+import java.lang.management.MemoryUsage;
+import java.util.Arrays;
+import java.util.List;
 
 public class JvmMemoryMetrics extends AbstractLifecycleListener {
 
+    public static final List<WildcardMatcher> TENURED_POOL_WILDCARDS = Arrays.asList(
+        WildcardMatcher.valueOf("CMS Old Gen"),
+        WildcardMatcher.valueOf("PS Old Gen"),
+        WildcardMatcher.valueOf("G1 Old Gen"),
+        WildcardMatcher.valueOf("Old Space"),
+        WildcardMatcher.valueOf("Tenured Gen"),
+        WildcardMatcher.valueOf("Java heap"),
+        WildcardMatcher.valueOf("GenPauseless Old Gen"));
+
     @Override
     public void start(ElasticApmTracer tracer) {
         bindTo(tracer.getMetricRegistry());
     }
 
     void bindTo(final MetricRegistry registry) {
         final MemoryMXBean platformMXBean = ManagementFactory.getPlatformMXBean(MemoryMXBean.class);
-        registry.add("jvm.memory.heap.used", Labels.EMPTY, new DoubleSupplier() {
-            @Override
-            public double get() {
-                return platformMXBean.getHeapMemoryUsage().getUsed();
-            }
-        });
-        registry.add("jvm.memory.heap.committed", Labels.EMPTY, new DoubleSupplier() {
-            @Override
-            public double get() {
-                return platformMXBean.getHeapMemoryUsage().getCommitted();
+
+        addUsedMetric("jvm.memory.heap.used", registry, platformMXBean.getHeapMemoryUsage());
+        addCommittedMetric("jvm.memory.heap.committed", registry, platformMXBean.getHeapMemoryUsage());
+        addMaxMetric("jvm.memory.heap.max", registry, platformMXBean.getHeapMemoryUsage());
+        addUsedMetric("jvm.memory.non_heap.used", registry, platformMXBean.getNonHeapMemoryUsage());
+        addCommittedMetric("jvm.memory.non_heap.committed", registry, platformMXBean.getNonHeapMemoryUsage());
+        addMaxMetric("jvm.memory.non_heap.max", registry, platformMXBean.getNonHeapMemoryUsage());
+
+        List<MemoryPoolMXBean> memoryPoolMXBeans = ManagementFactory.getMemoryPoolMXBeans();
+
+        for (MemoryPoolMXBean memoryPoolMXBean : memoryPoolMXBeans) {
+            if (memoryPoolMXBean.getType() != MemoryType.HEAP) {
+                continue;
             }
-        });
-        registry.add("jvm.memory.heap.max", Labels.EMPTY, new DoubleSupplier() {
-            @Override
-            public double get() {
-                return platformMXBean.getHeapMemoryUsage().getMax();
+            if (memoryPoolMXBean.getName().contains("Eden")) {
```

Comment:
I think it will be more informative. And this detailed information can help to add specific alerts.
Yes, I mean that they will be reported separately but with different lables(I fixed it with last commit)
![image](https://user-images.githubusercontent.com/48118512/84603108-ab9d2800-aead-11ea-9992-a046687a8003.png)